content/docs/rfcs/36/README.md

```diff
@@ -0,0 +1,738 @@
+
+`char *` containing the default pubsub topic:
+
+```
+/waku/2/default-waku/proto
+```
+
+### `extern char* waku_relay_publish(int nodeId, char* messageJson, char* pubsubTopic, int timeoutMs)`
+
+Publish a message using waku relay.
+
+**Parameters**
+
+1. `int nodeId`: The node identifier obtained from a successful execution of [`waku_new`](#extern-char-waku_newchar-configjson).
+2. `char* messageJson`: JSON string containing the [Waku Message](https://rfc.vac.dev/spec/14/) as [`JsonMessage`](#JsonMessage).
+3. `char* pubsubTopic`: pubsub topic on which to publish the message.
+   If `NULL`, it uses the default pubsub topic.
+4. `int timeoutMs`: Timeout value in milliseconds to execute the call.
+   If the function execution takes longer than this value,
+   the execution will be canceled and an error returned.
+   Use `0` for no timeout.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse).
+If the execution is successful, the `result` field contains the message ID.
+
+### `extern char* waku_relay_enough_peers(int nodeId, char* pubsubTopic)`
+
+Determine if there are enough peers to publish a message on a given pubsub topic.
+
+**Parameters**
+
+1. `int nodeId`: The node identifier obtained from a successful execution of [`waku_new`](#extern-char-waku_newchar-configjson).
+2. `char* pubsubTopic`: Pubsub topic to verify.
+   If `NULL`, it verifies the number of peers in the default pubsub topic.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse).
+If the execution is successful, the `result` field contains a `boolean` indicating whether there are enough peers.
+
+For example:
+
+```json
+{
+  "result": true
+}
+```
+
+### `extern char* waku_relay_subscribe(int nodeId, char* pubsubTopic)`
+
+Subscribe to a Waku Relay pubsub topic to receive messages.
+
+**Parameters**
+
+1. `int nodeId`: The node identifier obtained from a successful execution of [`waku_new`](#extern-char-waku_newchar-configjson).
+2. `char* pubsubTopic`: Pubsub topic to subscribe to. 
+   If `NULL`, it subscribes to the default pubsub topic.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse).
+If the execution is successful, the `result` field contains an `integer` that is the subscription id.
+
+For example:
+
+```json
+{
+  "result": 0
+}
+```
+
+**Events**
+
+When a message is received, a ``"message"` event` is emitted containing the message, pubsub topic, and node ID in which
+the message was received.
+
+The `event` type is [`JsonMessageEvent`](#jsonmessageevent).
+
+For Example:
+
+```json
+{
+  "nodeId": 1,
+  "type": "message",
+  "event": {
+    "subscriptionID": 1,
+    "pubsubTopic": "/waku/2/default-waku/proto",
+    "messageID": "0x6496491e40dbe0b6c3a2198c2426b16301688a2daebc4f57ad7706115eac3ad1",
+    "wakuMessage": {
+      "payload": "TODO",
+      "contentTopic": "/my-app/1/notification/proto",
+      "version": 1,
+      "timestamp": 1647826358000000000
+    }
+  }
+}
+```
+
+### `extern char* waku_relay_close_subscription(int nodeId, char* subscriptionId)`
```

Comment:
If the underlying effect is indeed to unsubscribe, shouldn't this be `waku_relay_unsubscribe`?

Reply:
Keen to rename but I think we need to clarify the behaviour. See other comments.

Reply:
Ok so @richard-ramos has simplified the API, does it mean we still need this specific function or we can just have `waku_relay_unsubscribe` ?